client/client_test.go

```diff
@@ -337,21 +337,6 @@ func TestHTTPClusterClientDo(t *testing.T) {
 			wantPinned: 1,
 		},
 
-		// context.DeadlineExceeded short-circuits Do
```

Comment:
Can we add a deadline reached test to ensure we return the deadline error correctly (not cluster error)